enhancements/rhcos/rhcos-cloud-marketplace-images.md

```diff
@@ -0,0 +1,330 @@
+---
+title: marketplace-images
+authors:
+  - "@patrickdillon"
+reviewers: # Include a comment about what domain expertise a reviewer is expected to bring and what area of the enhancement you expect them to focus on. For example: - "@networkguru, for networking aspects, please look at IP bootstrapping aspect"
+  - TBD
+approvers: # A single approver is preferred, the role of the approver is to raise important questions, help ensure the enhancement receives reviews from all applicable areas/SMEs, and determine when consensus is achieved such that the EP can move forward to implementation.  Having multiple approvers makes it difficult to determine who is responsible for the actual approval.
+  - "@sdodson"
+api-approvers: # In case of new or modified APIs or API extensions (CRDs, aggregated apiservers, webhooks, finalizers). If there is no API change, use "None"
+  - "None"
+creation-date: 2024-07-12
+last-updated: 2024-07-12
+tracking-link: # link to the tracking ticket (for example: Jira Feature or Epic ticket) that corresponds to this enhancement
+  - "SPSTRAT-271"
+see-also:
+replaces:
+superseded-by:
+---
+
+# RHCOS Cloud-Marketplace Images
+
+## Summary
+
+This enhancement proposes a process to upload/publish
+cloud (AWS, Azure, & GCP) marketplace images and include
+the image details in the RHCOS stream.
+
+## Motivation
+
+Publishing marketplace versions in the RHCOS stream will
+improve user experience and relieve tech debt:
+
+* Marketplace images published as part of Project Stratosphere are incomplete, requiring users to use
+boot images from previous releases, which slows
+down machine provisioning.
+* As marketplace images are not included in the coreos
+stream, users must manually enter the image details.
+With these changes, the experience will be improved
+so that users can simply indicate they want to use
+marketplace images and the installer will select the
+appropriate image.
+* Publishing a free Azure marketplace image would
+significantly decrease Azure installation times &
+would unblock Azure hosted control planes. On Azure, marketplace
+images are the only globally available, production-ready API for
+publishing images. Currently, every install creates a managed image
+utilizing the 16GB VHD file found in the RHCOS stream. Publishing
+an image would immediately save time during installs, reduce
+installer code maintenance, and reduce cluster cost.
+
+### User Stories
+
+* As an OpenShift cluster admin, I want to specify
```

Comment:
Ideally we switch this on by default and the user story is that new clusters are using marketplace images by default? The boolean should be a temporary situation?

Reply:
The intention here was to allow users to opt-in to using **paid** marketplace images. So for Azure we would default to using marketplace images with `NoPurchasePlan` and allow them to specify using a paid plan.

Reply:
So folks who pay for OCP through market places run the installer directly or through some cloud portal?

Reply:
We do have a documented install pattern for folks to run the installer directly and use marketplace workers.